src/Orckestra.Composer.Cart/Api/OrderController.cs

```diff
@@ -170,5 +170,76 @@ public virtual async Task<IHttpActionResult> GetGuestOrderByNumber(GetGuestOrder
 
             return Ok(viewModel);
         }
+
+        /// <summary>
+        /// Set an order in edit mode
+        /// </summary>
+        /// <param name="param">Parameters container</param>
+        [HttpPost]
+        [ActionName("edit-order")]
+        [ValidateModelState]
+        public virtual async Task<IHttpActionResult> EditOrder(EditOrderParam param)
+        {
+            if (param == null) return BadRequest($"{nameof(param)} cannot be empty");
+
+            var vm = await OrderHistoryViewService.CreateEditingOrderViewModel(param.OrderNumber).ConfigureAwait(false);
+
+            return Ok(vm);
+        }
+
+
+        /// <summary>
+        /// Cancel edit mode for order
+        /// </summary>
+        /// <param name="param">Parameters container</param>
+        [HttpPost]
+        [ActionName("cancel-edit-order")]
+        [ValidateModelState]
+        public virtual async Task<IHttpActionResult> CancelEditOrder(EditOrderParam param)
+        {
+            if (param == null) return BadRequest($"{nameof(param)} cannot be empty");
+
+            await OrderHistoryViewService.CancelEditingOrderAsync(param.OrderNumber).ConfigureAwait(false);
```

Comment:
The action has  [ValidateModelState] attribute and it will validate properties by rules
![image](https://user-images.githubusercontent.com/6649972/154214697-3b8224d5-1d07-4b32-a746-ed43b39e6b65.png)